### ISSUE TITLE
Is e2fslibs needed on MacOS X? - trying to use MacPorts

### ISSUE THREAD

**Issue by ajmas** [alperakcan/fuse-ext2#51]
I am in the process of trying to adapt the instructions to run with MacPorts on MacOS X, but 'configure' is failing because it can't find e2fslibs, yet this is not specified as a dependency in the current instructions for MacOS X.

Installation of dependencies:

```
sudo port install autoconf m4 automake libtool e2fsprogs
```

Failure I am getting when trying to run ``configure``:

```
checking for strerror... yes
checking for strrchr... yes
checking for strtol... yes
checking for strtoul... yes
checking for strtoull... yes
checking for uname... yes
checking for utime... yes
checking for llseek... no
checking for lseek64... no
checking for library containing sem_post... none required
checking for library containing fuse_main... -losxfuse
checking for library containing com_err... -lcom_err
checking for library containing ext2fs_open... no
configure: error: Can't find e2fslibs, please install it
```

Looking at the library ``/opt/local/lib/libext2fs.a`` with ``strings``, I see:

```
_ext2fs_open_inode_scan
_ext2fs_open
_ext2fs_open2
_ext2fs_open_file
```

Envirornment settings:

```
PKG_CONFIG_PATH=/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/opt/X11/bin:/usr/texbin:/Users/ajmas/npm/bin:/opt/local/bin
PKG_CONFIG_PATH=/opt/local/lib/pkgconfig:/opt/local/lib/pkgconfig:/usr/lib/pkgconfig  
CFLAGS="-idirafter/opt/local/include -idirafter/usr/local/include -idirafter/usr/local/include/osxfuse/"
LDFLAGS="-L/opt/local/lib -L/usr/local/lib"
```

**Comment by alperakcan** [comment]
what abount config.log file.

**Comment by danormsby** [comment]
Same issue.  Config.log file as a zip archive attached.

[config.zip](https://github.com/alperakcan/fuse-ext2/files/682045/config.zip)

**Comment by riktov** [comment]
Are you using the recommended configure command line?

`CFLAGS="-idirafter/opt/gnu/include -idirafter/usr/local/include/osxfuse/" LDFLAGS="-L/opt/gnu/lib -L/usr/local/lib" ./configure`

And replace all the /opt/gnu and /usr/local paths with /opt/local. This worked for me; it made the missing e2fslibs error go away I was able to build. Though I can't get fuse-ext2 to mount the disk.

**Comment by Ole2mail** [comment]
ext2fs_open is inside e2fsprogs which is available to install from homebrew with the following command:

brew install e2fsprogs

but the configure will fail to find the libext2fs.a on a standard path so the only following one worked for me:

CFLAGS="-idirafter/opt/gnu/include -idirafter/usr/local/include/osxfuse/ -idirafter/$(brew --prefix e2fsprogs)/include" LDFLAGS="-L/usr/local/opt/glib -L/usr/local/lib -L$(brew --prefix e2fsprogs)/lib" ./configure

**Comment by ylluminarious** [comment]
I was able to get `fuse-ext2` to compile just fine with Macports. All I needed was to install these ports:

```sh
# port install e2fsprogs
# port install libtool
```

And then I compiled like so:

```sh
$ CFLAGS="-I/opt/local/include" LDFLAGS="-L/opt/local/lib" ./configure
$ make
```

Then do `make install`, etc. to install, although this varies if you want the System Preferences prefpane.

**Comment by llamafilm** [comment]
The required files are included in `e2fsprogs` from MacPorts.  This configure command worked for me.  The command above from ylluminarious didn't work for me because my fuse is installed in /usr/local/lib not from MacPorts.

```
CFLAGS="-idirafter/opt/local/include -idirafter/usr/local/include" LDFLAGS="-L/opt/local/lib -L/usr/local/lib" ./configure
```

**Comment by ylluminarious** [comment]
@llamafilm Yes, sorry, sometimes you need to use different paths depending on how you have things set up. I'm going to try and get this program into MacPorts in the not-too-distant future to ease the installation process, since we were rejected by Homebrew. Hopefully that will eliminate issues like this and one can simply `sudo port install ext2fuse`.
